course6/toyapp-adam/util/mouse.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include "pixel.h"
+
+namespace Util {
+
+class Mouse
+{
+public:
+	Pixel GetCursorPosition() const { return mCursorPosition; }
+	Pixel GetCursorPositionDelta() const { return mCursorPositionDelta; }
+
+	void AddDeltaPosition(Pixel delta)
+	{
+		mCursorPositionDelta.x += delta.x;
+		mCursorPositionDelta.y += delta.y;
+	}
+
+	void RecordPosition()
```

Comment:
This API seems to be very bug-prone. You have 2, order-dependent state modifiers. Seems complex for sucha  simple thing.

Reply:
Sorry I'm not sure where you're getting at. Why should this Mouse worry about what order its methods are called, why is this a bad sign? Reason for this method is the same as keyboard's pressed and down states. The state must be cleared between simulation Updates.

I accumulate SDL mouse move events, since as far as I know more than one can happen between updates. Unless SDL does some kind of magic.